lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -357,17 +357,9 @@ static bool writesToAccumulator(Operation *op) {
         .getElementType()
         .isa<IntegerType>();
   } else if (auto fma_elemOp = dyn_cast<aievec::FMAElemOp>(op)) {
-    return fma_elemOp.getResult()
-        .getType()
-        .cast<VectorType>()
-        .getElementType()
-        .isa<IntegerType>();
+    return true;
   } else if (auto mul_elemOp = dyn_cast<aievec::MulElemOp>(op)) {
-    return mul_elemOp.getResult()
-        .getType()
-        .cast<VectorType>()
-        .getElementType()
-        .isa<IntegerType>();
+    return true;
   } else if (isa<aievec::UPSOp>(op))
     return true;
```

Comment:
You can merge all of this in a single "if" statement with a multiple "isa" check. Something like "isa<FMAElemOp, MulElemOp, UPSOp>(op)".